.github/workflows/run-pylint.yml

```diff
@@ -0,0 +1,18 @@
+name: "Check python code"
+
+on: [push, pull_request]
+
+jobs:
+  using-pylint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install pylint and action deps
+        run: |
+          python3 -m pip install --upgrade pylint
+          python3 -m pip install -r requirements.txt
+      - name: run pylint
+        run: pylint python_action/**
```

Comment:
`pylint python_action/**`  seems not include `setup.py` under root folder. `pylint *.py` can include all python files.

Reply:
I usually don't pylint setup.py files because they tend to have some cases that should be ignored. But I will add it with something like
```suggestion
        run: |
          pylint python_action/**
          pylint setup.py
```

Reply:
I see you're using gitpod. I also think gitpod is awesome! If you want I can add this badge to the README

![[open repo in gitpod](https://gitpod.io/#https://github.com/shenxianpeng/cpp-linter-action)](https://img.shields.io/badge/Gitpod-Use%20Online%20IDE-B16C04?logo=gitpod)

Reply:
Okay, please add it. Actually, I‘d like to ask you what is your debugging method, it's hard to debug GitHub action, so you also use gitpod to develop?

Reply:
>you also use gitpod to develop

no, I use gitpod to browse code when I'm away from my PC. On PC, I use VSCode (which was recently adopted by gitpod - gitpod used to use atom or something similar)..

>I‘d like to ask you what is your debugging method

If look at the source code, you'll see that I hard-coded some environment variables' values.
```py
# info from my current check-python-sources repo's run.py script
GITHUB_EVEN_PATH = os.getenv("GITHUB_EVENT_PATH", "event_payload.json")
GITHUB_API_URL = os.getenv("GITHUB_API_URL", "https://api.github.com")
GITHUB_REPOSITORY = os.getenv("GITHUB_REPOSITORY", "2bndy5/check-python-sources")
GITHUB_EVENT_NAME = os.getenv("GITHUB_EVENT_NAME", "push")
GITHUB_RUN_ID = os.getenv("GITHUB_RUN_ID", "0")
GITHUB_SHA = os.getenv("GITHUB_SHA", "cf989be19c74ad1d27614badca3bc7f63eb73806")
GITHUB_TOKEN = os.getenv("GITHUB_TOKEN", os.getenv("GIT_REST_API", None))
API_HEADERS = {
    "Authorization": f"token {GITHUB_TOKEN}",
    "Accept": "application/vnd.github.v3.json",
}
```
this lets me debug a single event locally with variables that would normally only be available when executed on the github runner.

Additionally, I also keep a file called _event_payload.json_ (it is in the _.gitignore_). This file essentially contains the same info from a failed run on github. I have the action's logging verbosity set to `10` (debug level) so I can copy the event's payload from the run's log output into my local _event_payload.json_ file.

Lastly, to use the REST API without having to know a repo's secret token, I have created a personal token, and saved it to my PC's environment variable `GIT_REST_API`. In the above snippet, you'll notice that I have the `GITHUB_TOKEN` set to use my `GIT_REST_API` variable.

I don't use docker at all (for speed), but I do tend to use Windows PS and WSL Ubuntu terminal to run
```
python3 -m python_action.run --verbosity=20
```
I do make sure that any dependent software/libraries are installed (typically in a python virtual environment when applicable). Note that the `python_action` pkg doesn't need to be installed when executing from the cloned repo's root directory.

Reply:
I like to use python the pkg `rich` for colored logging and human readable JSON output in the logs. Try it out:
```
python3 -m pip install rich
```
You'll notice that the _python_action.__init__.py_ looks for this pkg and fails gracefully
```py
FOUND_RICH_LIB = False
try:
    from rich.logging import RichHandler

    FOUND_RICH_LIB = True

    logging.basicConfig(
        format="%(name)s: %(message)s",
        handlers=[RichHandler(show_time=False)],
    )

except ImportError:
    logging.basicConfig()

#: The logging.Logger object used for outputing data.
logger = logging.getLogger("Python-Checker")
if not FOUND_RICH_LIB:
    logger.debug("rich module not found")
```